add object_ids parameter to role_user_assignment

adonisgarciac opened the pull request:
<!-- markdownlint-disable -->
# What does this PR do?
adding object_ids to gateway_role_user_assignments

adonisgarciac committed:
add object_ids parameter to role_user_assignment

ivarmu reviewed (changes requested):
As `object_id` is deprecated, shouldn't it be also be reflected here?
```
   object_id  Deprecated
              This option is deprecated and will be removed in a release after 2026-01-31.
              For associating a user to team(s)/organization(s), please use the object_ids param.
               -------------
              Primary key/Name of the object this assignment applies to.
              This option is mutually exclusive with object_ids and object_ansible_id.
        default: null
        type: int

   object_ids  List of object IDs(Primary Key ) or names this assignment applies to.
               This option is mutually exclusive with object_id and object_ansible_id.
        default: null
        elements: str
        type: list
```

Also, the documentation may be updated and a changelog fragment is needed.

adonisgarciac committed:
add option to readme and changelog

adonisgarciac commented:
I've added it to README and changelog. I wouldn't remove object_id even though it is deprecated until deprecation is effective. I added deprecation warning to README

ivarmu reviewed (approved):
LGTM
